### DISCUSSION TITLE
Protect pets in God mode (with example code).

### DISCUSSION THREAD

**Discussion by Nicholamsious** [EssentialsX/Essentials#4522] [Ideas and Feature Suggestions] [answered]
# The problem
A player's pets can currently receive damage from other players and/or their pets while their owner is in god mode.

# The idea
To protect pets that belong to a player from damage (from other players or their pets) if the player is in god mode.
### But a small problem is...
Only pets that belong to an online player can be protected with the code I have created. If the user of an offline player can be fetched if they stay in god mode offline, that's worth changing.

## An example
I have already[ written code for this idea](https://github.com/nsgwick/Essentials/blob/2.x/Essentials/src/main/java/com/earth2me/essentials/EssentialsEntityListener.java#L104-L174), so here are some screenshots of it in action:

![2021-09-12_12 17 46](https://user-images.githubusercontent.com/90180047/132987579-938a682d-b9f9-419a-99ab-0682066adad6.png)
![2021-09-12_12 18 16](https://user-images.githubusercontent.com/90180047/132987578-1000ee7f-1845-496b-8e4d-b09ce2c0d0f6.png)

Test build: [click here](https://github.com/nsgwick/Essentials/releases/tag/2.13.1)

# This would probably need:
- a couple of new config options for pet protection if it would be added.
- expansion to other parts of the code if not only for god mode

**Comment by oskarkk** [comment]
Great idea. This could be also a separate command to protect all your pets regardless of god mode.

**Comment by mdcfe** [comment, marked as answer]
Going to echo my concerns from #4533 here:

I appreciate the effort that has gone into the development of the pet protection features in #4533, but I remain unconvinced that this is something we would want to include in EssentialsX itself. Some of the ideas in the PR would work better as a standalone plugin.

That said, I wouldn't be completely opposed to a config option to protect pets while in god mode. I would be willing to consider a PR with a smaller scope that covers god mode alone, as suggested in this original issue.
